src/bn256/mod.rs

```diff
@@ -141,6 +141,21 @@ pub struct PrivateKey(bn::Fr);
 pub struct PublicKey(bn::G2);
 
 impl PrivateKey {
+
+    /// Function to derive a private key.
+    fn new(rng: &[u8]) -> Result<PrivateKey, Error> {
+
+        // This function throws an error if the slice does not have a proper length.
+        let private_key = Fr::from_slice(&rng)?;
+
+        // From_slice does not check whether we specified an all 0 slice
+        if private_key == Fr::zero() {
```

Comment:
Actually, I haven't found any sources that say the private key cannot be zero, but some implementations check that it is lower than the curve order.

Reply:
Obviously from a field perspective it can be possible. The questions is whether we want to allow it. 

As for the curve order, same applies. it does not really matter as you will still generate a valid point. Depends on what we want to do with those cases